upgrade/v1beta1/upgrade.go

```diff
@@ -120,6 +139,43 @@ func upgradeVolumes(namespace string, lhClient *lhclientset.Clientset) error {
 	return nil
 }
 
+func fixupVolumes(namespace string, lhClient *lhclientset.Clientset) error {
+	volumes, err := lhClient.LonghornV1beta1().Volumes(namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: getLabelSelectorNotEqualV1beta2APIVersion(),
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, obj := range volumes.Items {
+		if obj.Spec.DiskSelector == nil {
+			obj.Spec.DiskSelector = []string{}
+		}
+		if obj.Spec.NodeSelector == nil {
+			obj.Spec.NodeSelector = []string{}
+		}
+		if obj.Spec.RecurringJobs == nil {
+			obj.Spec.RecurringJobs = make([]longhornV1beta1.VolumeRecurringJobSpec, 0)
+		}
+		for i, src := range obj.Spec.RecurringJobs {
+			dst := longhornV1beta1.VolumeRecurringJobSpec{}
+			copier.Copy(&dst, &src)
+			if dst.Groups == nil {
+				dst.Groups = []string{}
+			}
+			if dst.Labels == nil {
+				dst.Labels = make(map[string]string, 0)
+			}
+			obj.Spec.RecurringJobs[i] = dst
+		}
+
+		if _, err = lhClient.LonghornV1beta1().Volumes(namespace).Update(context.TODO(), &obj, metav1.UpdateOptions{}); err != nil {
```

Comment:
Should we call `LonghornV1beta2` update?

Reply:
Hmm... The Status.Conditions are not converted yet, and the Update operation also validates the Status.
Thus, using v1beta1.Update is the only way to update Spec.

Reply:
I see.